### PR TITLE
Add NimBLEClientCallbacks::onConnectFail callback

### PR DESCRIPTION
**Pull request by h2zero**
Adds a callback that is called when the connection attempt fail while connecting asynchronously.

### CODE REVIEW

**Comment by h2zero** [comment]
@thekurtovic Please let me know if this works for you.

File: src/NimBLEClient.cpp [hunk 1]
```diff
@@ -989,15 +989,16 @@ int NimBLEClient::handleGapEvent(struct ble_gap_event* event, void* arg) {
...
+                if (pClient->m_config.deleteOnConnectFail) {
                     delete pClient;
-                    return 0;
                 }
+                return 0;
             }
 
-            if (pClient->m_config.asyncConnect) {
-                pClient->m_pClientCallbacks->onConnect(pClient);
-            } else if (!pClient->m_config.exchangeMTU) {
```
Comment (thekurtovic) [inline src/NimBLEClient.cpp:1000]: This call to `onConnect` should be left as is for async connects.

Comment (h2zero) [reply src/NimBLEClient.cpp:1000]: Oops, I actually meant to move that instead of delete, thanks!

**Comment by thekurtovic** [comment]
LGTM. Tested with esp32-s3 and it resolves recent issues mentioned in https://github.com/h2zero/esp-nimble-cpp/issues/232.

**Pull request merged**